### PR TITLE
Adding Permission Boundary option to IAM Role Creation

### PR DESCRIPTION
**Pull request by gthomson31**
#3  - This PR is to fix the missing permissions boundary

### CODE REVIEW

File: cluster/main.tf [hunk 1]
```diff
@@ -45,11 +45,15 @@ data "aws_iam_policy_document" "doit_eks_lens_collector" {
   }
 }
 
+
```
Comment (eyalzek) [inline cluster/main.tf]: remove extra line

File: cluster/main.tf [hunk 2]
```diff
@@ -45,11 +45,15 @@ data "aws_iam_policy_document" "doit_eks_lens_collector" {
...
 resource "aws_iam_role" "doit_eks_lens_collector" {
   count = var.ec2_cluster ? 0 : 1
 
   name               = "doit_eks_${local.region}_${var.cluster.name}"
   assume_role_policy = data.aws_iam_policy_document.doit_eks_lens_collector[count.index].json
+
+  permissions_boundary = var.permissions_boundary
+
```
Comment (eyalzek) [inline cluster/main.tf]: remove extra line

File: region-base/main.tf [hunk 3]
```diff
@@ -67,6 +67,13 @@ data "aws_iam_policy_document" "doit_eks_lens_import" {
 resource "aws_iam_role" "doit_eks_lens_import" {
   name               = "doitintl_eks_import_${local.region}"
   assume_role_policy = data.aws_iam_policy_document.doit_eks_lens_import.json
+
+  tags = {
+    Name = "doitintl_eks_import_${local.region}"
+  }
+
+  permissions_boundary = var.permissions_boundary
+
```
Comment (eyalzek) [inline region-base/main.tf]: remove extra line

**Review by eyalzek — CHANGES_REQUESTED** [review]
Requested some formatting changes.

Could you also give an example of a value to be passed to this new variables? I'm not familiar with this feature.

**Comment by gthomson31** [comment]
> Requested some formatting changes.
> 
> Could you also give an example of a value to be passed to this new variables? I'm not familiar with this feature.

WIll make the changes suggested - We use a permissions boundaries to set a limitation on the permissions any role or user can have.

Example would be passing an existing AWS policy ARN to be used as the boundary

By default any of our users must attach the permissions boundary to any created IAM Resource/.
This way we can have a more granilur control over what our users and roles have access to within our estate that couldnt be covered by SCP.

I have posted an example boundary policy below.

https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html

```

{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Action": "*",
            "Resource": "*",
            "Effect": "Allow",
            "Sid": "AllowAllAction"
        },
        {
            "Condition": {
                "StringNotEquals": {
                    "iam:PermissionsBoundary": "arn:aws:iam::123456789012:policy/ExampleBoundaryPolicy"
                }
            },
            "Action": [
                "iam:AttachRolePolicy",
                "iam:CreateRole",
                "iam:DeleteRolePolicy",
                "iam:DetachRolePolicy",
                "iam:PutRolePermissionsBoundary",
                "iam:AttachUserPolicy",
                "iam:CreateUser",
                "iam:DeleteUserPolicy",
                "iam:DetachUserPolicy",
                "iam:PutUserPermissionsBoundary"
            ],
            "Resource": "*",
            "Effect": "Deny",
            "Sid": "AllowCreationOfRolesWithOutBoundary"
        },
        {
            "Action": [
                "iam:*Put*",
                "iam:*Create*",
                "iam:*Delete*"
            ],
            "Resource": [
                "arn:aws:iam::123456789012:policy/ExampleBoundaryPolicy"
            ],
            "Effect": "Deny",
            "Sid": "DenyModificationBoundaryPolicy"
        },
        {
            "Action": [
                "iam:CreatePolicyVersion",
                "iam:DeletePolicy",
                "iam:DeletePolicyVersion",
                "iam:SetDefaultPolicyVersion"
            ],
            "Resource": [
                "arn:aws:iam::123456789012:policy/ExampleBoundaryPolicy"
            ],
            "Effect": "Deny",
            "Sid": "DenySpecificActionForBoundaryPolicy"
        }
    ]
}

File: cluster/main.tf [hunk 4]
```diff
@@ -50,6 +50,7 @@ resource "aws_iam_role" "doit_eks_lens_collector" {
 
   name               = "doit_eks_${local.region}_${var.cluster.name}"
   assume_role_policy = data.aws_iam_policy_document.doit_eks_lens_collector[count.index].json
+  permissions_boundary = var.permissions_boundary
```
Comment (hobord) [inline cluster/main.tf:53]: align =

File: region-base/main.tf [hunk 5]
```diff
@@ -67,6 +67,12 @@ data "aws_iam_policy_document" "doit_eks_lens_import" {
 resource "aws_iam_role" "doit_eks_lens_import" {
   name               = "doitintl_eks_import_${local.region}"
   assume_role_policy = data.aws_iam_policy_document.doit_eks_lens_import.json
+  permissions_boundary = var.permissions_boundary
```
Comment (hobord) [inline region-base/main.tf:70]: align =

**Review by hobord — CHANGES_REQUESTED** [review]

**Pull request merged**